xbmc/windowing/gbm/DRMUtils.cpp

```diff
@@ -197,6 +197,31 @@ bool CDRMUtils::SupportsProperty(struct drm_object *object, const char *name)
   return false;
 }
 
+bool CDRMUtils::SupportsPropertyAndValue(struct drm_object* object,
+                                         const char* name,
+                                         uint64_t value /* = 0 */)
+{
+  for (uint32_t i = 0; i < object->props->count_props; i++)
+  {
+    if (!strcmp(object->props_info[i]->name, name))
+    {
+      if (drm_property_type_is(object->props_info[i], DRM_MODE_PROP_ENUM) != 0)
+      {
+        for (int j = 0; j < object->props_info[i]->count_enums; j++)
+        {
+          if (object->props_info[i]->enums[j].value == value)
+            return true;
```

Comment:
Sometimes deep nesting can make the logic less readable. Nesting can be reduced by inverting the `if` logic and adding a `continue`, e.g.

```c++
{
  for (uint32_t i = 0; i < object->props->count_props; i++)
  {
    if (strcmp(object->props_info[i]->name, name) != 0)
      continue;

    if (drm_property_type_is(object->props_info[i], DRM_MODE_PROP_ENUM) == 0)
      continue;

    for (int j = 0; j < object->props_info[i]->count_enums; j++)
    {
      if (object->props_info[i]->enums[j].value == value)
        return true;
```

Reply:
Also `strcmp` is one of those gotchya functions and the source of many a bug, so make its comparison as clear as f*ing day

Reply:
thanks! I want to continue nesting the `drm_property_type_is` check as it's possible in the future to add more checks for different types as we need them ie. `DRM_MODE_PROP_RANGE` and `DRM_MODE_PROP_SIGNED_RANGE`.

as for strcmp I can always change all the usage of them to `StringUtils::CompareNoCase()` I guess but I'm not sure if there is any benefit :smile:

Reply:
I would recommend changing to CompareNoCase(), unless the performance impact is measurable then this provides much better clarity and features that desired boolean return.

Reply:
done :)